games-fps/gzdoom/gzdoom-4.1.2.ebuild

```diff
@@ -0,0 +1,65 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils desktop xdg
+
+DESCRIPTION="A modder-friendly OpenGL source port based on the DOOM engine"
+HOMEPAGE="https://zdoom.org"
+SRC_URI="https://github.com/coelckers/${PN}/archive/g${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD BZIP2 DUMB-0.9.2 GPL-3 LGPL-3 MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="fluidsynth gtk openal openmp"
+
+DEPEND="
+	media-libs/libsdl2[opengl]
+	sys-libs/zlib
+	virtual/jpeg:0
+	gtk? ( <x11-libs/gtk+-4:= )"
```

Comment:
GZDoom supports GTK+ version 2 and version 3. In order to allow the ebuild to support both slots, I made the dependency be `<x11-libs/gtk+-4:=`. The reason for `=` is to cause a rebuild if the user switches slots between `gtk+:3` and `gtk+:2`.

I can remove support for the `gtk+:2` slot if you think that would be better; for what it's worth, I don't think much of the GZDoom team is using GTK+ version 2 anymore, despite support for it still in the code. Is `gtk+:2` deprecated for Gentoo packages?

Reply:
I see. `=` forces rebuild if a subslot is changed. It is not needed, gtk+ has no subslots at all, it will only slow down portage.
If gtk+-3 is the primary version now, but gtk+-2 is supported then add a local USE flag `gtk2`:
```
gtk? (
!gtk2? ( x11-libs/gtk+:3 )
gtk2? ( x11-libs/gtk+:2 )
)
```
(or USE=gtk3 if otherwise)